homeassistant/components/squeezebox/coordinator.py

```diff
@@ -108,6 +140,18 @@ async def _async_update_data(self) -> dict[str, Any]:
                 self._remove_dispatcher = async_dispatcher_connect(
                     self.hass, SIGNAL_PLAYER_REDISCOVERED, self.rediscovered
                 )
+
+            elif self.player.alarms:
+                for alarm in self.player.alarms:
+                    if alarm["id"] not in self.known_alarms:
+                        self.known_alarms.append(alarm["id"])
+                        async_dispatcher_send(
+                            self.hass, SIGNAL_ALARM_DISCOVERED, alarm, self
+                        )
+                alarm_dict: dict[str, Alarm] = {
+                    alarm["id"]: alarm for alarm in self.player.alarms
+                }
+                return {"alarms": alarm_dict}
```

Comment:
How many more platforms will you add for this? Otherwise consider just creating a coordinator listener over a dispatcher